src/snowflake/cli/api/commands/flags.py

```diff
@@ -485,7 +485,7 @@ def project_definition_option(project_name: str):
 
     def _callback(project_path: Optional[str]):
         dm = DefinitionManager(project_path)
-        project_definition = getattr(dm.project_definition, project_name, None)
+        project_definition = dm.project_definition
```

Comment:
This change alters what is provided by `@with_project_definition` decorator. Instead of part of project definition (snowpark, streamlit, native_app) it provides full definition.

The only purpose of keeping `@with_project_definition(NAME)` is to add `-p/--project` with "nice" description that relates to project type.

However, we already have some name mapping:
https://github.com/snowflakedb/snowflake-cli/blob/532d1f10a8be36105ec689ce597d20ac6bbd8e50/src/snowflake/cli/api/commands/flags.py#L500-L505

So my suggestion would be to get rid of this "nice description" thing in favor of simplicity. In this way we can move the functionality of this decorator to `app.command(uses_project=True)`.

@sfc-gh-cgorrie @sfc-gh-pjob wdyt?

Reply:
should uses_project=True be the default?

Reply:
I don't think so, [explicite is better than implicite](https://peps.python.org/pep-0020/#the-zen-of-python). Not all command require access to project file. Imho the decorators should inform developer what is used by the command  instead of expecting them to know default values and behaviors. WDYT?

Reply:
IMO it looks good :)